GitUI/BranchTreePanel/RepoObjectsTree.cs

```diff
@@ -155,19 +155,19 @@ public async Task ReloadAsync()
             var currentBranch = Module.GetSelectedBranch();
             await this.SwitchToMainThreadAsync();
 
-            var token = CancelBackgroundTasks();
             Enabled = false;
 
             try
             {
+                var token = CancelBackgroundTasks();
                 treeMain.BeginUpdate();
                 _rootNodes.ForEach(t => t.IgnoreSelectionChangedEvent = true);
                 var tasks = _rootNodes.Select(r => r.ReloadAsync(token)).ToArray();
                 await Task.WhenAll(tasks).ConfigureAwait(false);
             }
             finally
             {
-                await this.SwitchToMainThreadAsync(token);
+                await this.SwitchToMainThreadAsync();
                 Enabled = true;
 
                 var selectedNode = treeMain.AllNodes().FirstOrDefault(n =>
```

Comment:
The token protection has to only be excluded for `EndUpdate` the rest has to remain unchanged.

```csharp

        public async Task ReloadAsync()
        {
            var currentBranch = Module.GetSelectedBranch();
            await this.SwitchToMainThreadAsync();

            var token = CancelBackgroundTasks();
            Enabled = false;

            try
            {
                treeMain.BeginUpdate();
                var tasks = _rootNodes.Select(r => r.ReloadAsync(token)).ToArray();
                await Task.WhenAll(tasks).ConfigureAwait(false);
            }
            finally
            {
                await this.SwitchToMainThreadAsync();
                if (!token.IsCancellationRequested)
                {
                    Enabled = true;
                    var selectedNode = treeMain.AllNodes().FirstOrDefault(n =>
                        _rootNodes.Any(rn => $"{rn.TreeViewNode.FullPath}{treeMain.PathSeparator}{currentBranch}" == n.FullPath));
                    if (selectedNode != null)
                    {
                        _rootNodes.ForEach(t => t.IgnoreSelectionChangedEvent = true);
                        treeMain.SelectedNode = selectedNode;
                        _rootNodes.ForEach(t => t.IgnoreSelectionChangedEvent = false);
                        treeMain.SelectedNode.EnsureVisible();
                    }
                }

                treeMain.EndUpdate();
            }
        }
```

Reply:
Yes, that makes sense @jbialobr. At first I wondered if we should include Enabled = true next to treeMain.EndUpdate, but I suppose it's fine since cancelling means another update is in flight. So during the time between cancelling and the next update, we'll see the tree's contents disabled. I'll upload an amendment with this version.